### DISCUSSION TITLE
MySQL-cdc同步到StarRocks，如果MySQL配置debezium.column.exclude.list=aa.order.price，StarRocks端会报错

### DISCUSSION THREAD

**Discussion by night-create** [apache/flink-cdc#3736] [Q&A] [answered]
<img width="1420" alt="6fca7ee1a6f64786badd5b1427c61a74" src="https://github.com/user-attachments/assets/8926d5f7-1c22-41ef-8a71-5f4cf7a6fa34">
报错的地方如下：
<img width="878" alt="Snipaste_2024-11-15_14-49-01" src="https://github.com/user-attachments/assets/f50df918-f19e-4a79-8622-21c805012cfe">
如果StarRocks库里没有order表，是通过cdc自动创建的，创建的时候并没有过滤price字段，而获取的数据里过滤了，导致131行进行检查时发现不相等导致的问题。这边的检查也是有必要的，毕竟后续streamload阶段如果字段数不相等也会报错。是否可以实现建表也能够过滤字段？

**Comment by yuxiqian** [comment, marked as answer]
`debezium.column.exclude.list` is known to be incompatible in Pipeline mode. Please use `transform.projection` to filter out columns instead.

**Reply by night-create** [reply]
ths
